crates/subspace-farmer/src/dsn.rs

```diff
@@ -82,7 +82,11 @@ where
                     PieceIndexHashNumber::one()
                 },
             )?;
-            let offset_end = ((i + 1) * range_size).min(sync_sector_size);
+            let (mut offset_end, overflow) = (i + 1).overflowing_mul(range_size);
```

Comment:
I think what you want is `saturating_mul`, `overflowing_mul` can often be replaced with a better alternative.